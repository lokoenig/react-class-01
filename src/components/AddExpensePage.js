import React from "react";
import { connect } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";


const AddExpensePage = (props) => {
  const navigate = useNavigate();
  return (
  <div>
    <ExpenseForm
        onSubmit={(expense) => {
          props.onSubmit(expense);
          navigate('/');
        }
        }
    />
  </div>
)};

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (expense) => dispatch(addExpense(expense))
  });

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
