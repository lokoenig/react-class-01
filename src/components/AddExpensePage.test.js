import React from "react";
import AddExpensePage from "./AddExpensePage";