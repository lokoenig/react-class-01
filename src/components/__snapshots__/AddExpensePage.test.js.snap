// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders full Edit Expense page 1`] = `
<DocumentFragment>
  <div>
    <h1>
      I be a form!
    </h1>
    <form
      title="Edit An Expense Entry"
    >
      <div
        class="form-input-group"
      >
        <div
          class="form-input field-wide"
        >
          <label
            for="expense-description"
          >
            Description
          </label>
          <input
            id="expense-description"
            name="description"
            placeholder="Description"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-input"
        >
          <label
            for="expense-amount"
          >
            Expense Amount
          </label>
          <input
            id="expense-amount"
            name="amount"
            placeholder="Amount"
            type="text"
            value="0"
          />
        </div>
        <div
          class="form-input"
        >
          <label
            for="form-datepicker"
          >
            Date
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <span
                aria-live="polite"
                class="react-datepicker__aria-live"
                role="alert"
              />
              <input
                class=""
                id="form-datepicker"
                type="text"
                value="02/21/2023"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="form-input-group"
      >
        <div
          class="form-input field-wide"
        >
          <label
            for="expense-note"
          >
            Note
          </label>
          <textarea
            id="expense-note"
            name="note"
            placeholder="Note"
          />
        </div>
        <div
          class="form-submit"
        >
          <button>
            Add Expense
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
